libs/core/src/components/sage-checkbox/sage-checkbox.tsx

```diff
@@ -0,0 +1,106 @@
+import { Component, h, Prop, Host } from '@stencil/core';
+
+@Component({
+  tag: 'sage-checkbox',
+  styleUrl: 'sage-checkbox.scss',
+  shadow: true,
+})
+export class SageCheckbox {
+  /**
+   * Whether or not the checkbox is checked.
+   */
+  @Prop() checked: boolean;
+
+  /**
+   * Whether or not the checkbox is disabled.
+   */
+  @Prop() disabled: boolean;
+
+  /**
+   * Whether or not the checkbox is invalid.
+   */
+  @Prop() invalid: boolean;
+
+  /**
+   * String used for checkbox `id` attribute and label `for` attribute.
+   */
+  @Prop() checkboxId: string;
+
+  /**
+   * String used for label text next to checkbox
+   */
+  @Prop() label: string;
+
+  /**
+   * String used for message below checkbox
+   */
+  @Prop() message: string;
+
+  /**
+   * String used for checkbox `name` attribute.
+   */
+  @Prop() name: string;
+
+  /**
+   * Whether or not checkbox should display as indeterminate. Prop is for visual styling only.
+   * Only JavaScript can set the objects `indeterminate` property. See [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#indeterminate_state_checkboxes)
```

Comment:
Could be  a good link to add at the end of the story in an `Additional Resources` section